// @mantine/core/cjs/Slider/Marks/Marks.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styles = require('@mantine/styles');
var SliderRoot_styles = require('../SliderRoot/SliderRoot.styles.js');

var useStyles = styles.createStyles((theme, { color, disabled }, { size }) => ({
  marksContainer: {
    position: "absolute",
    right: styles.getSize({ sizes: SliderRoot_styles.sizes, size }),
    left: styles.getSize({ sizes: SliderRoot_styles.sizes, size })
  },
  markWrapper: {
    position: "absolute",
    top: `calc(${styles.rem(styles.getSize({ sizes: SliderRoot_styles.sizes, size }))} / 2)`,
    zIndex: 2,
    height: 0
  },
  mark: {
    boxSizing: "border-box",
    border: `${styles.rem(2)} solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    height: styles.getSize({ sizes: SliderRoot_styles.sizes, size }),
    width: styles.getSize({ sizes: SliderRoot_styles.sizes, size }),
    borderRadius: 1e3,
    transform: `translateX(calc(-${styles.getSize({ sizes: SliderRoot_styles.sizes, size })} / 2))`,
    backgroundColor: theme.white,
    pointerEvents: "none"
  },
  markFilled: {
    borderColor: disabled ? theme.colorScheme === "dark" ? theme.colors.dark[3] : theme.colors.gray[4] : theme.fn.variant({ variant: "filled", color }).background
  },
  markLabel: {
    transform: `translate(-50%, calc(${theme.spacing.xs} / 2))`,
    fontSize: theme.fontSizes.sm,
    color: theme.colorScheme === "dark" ? theme.colors.dark[2] : theme.colors.gray[6],
    whiteSpace: "nowrap",
    cursor: "pointer",
    userSelect: "none"
  }
}));

exports.default = useStyles;
